preprocessing/structs.ts

```diff
@@ -1,15 +1,82 @@
 /* # Mangrove Summary
-   * The Mangrove holds order books for `outbound_tkn`,`inbound_tkn` pairs.
-   * Offers are sorted in a doubly linked list.
+   * Mangrove holds offer lists for `outbound_tkn`,`inbound_tkn` pairs with a given `tickSpacing`.
+   * Offers are sorted in a tree (the "tick tree") where each available price point (a `bin`) holds a doubly linked list of offers.
    * Each offer promises `outbound_tkn` and requests `inbound_tkn`.
    * Each offer has an attached `maker` address.
-   * In the normal operation mode (called Mangrove for Maker Mangrove), when an offer is executed, we:
+   * When an offer is executed, Mangrove does the following:
      1. Flashloan some `inbound_tkn` to the offer's `maker`.
      2. Call an arbitrary `execute` function on that address.
      3. Transfer back some `outbound_tkn`.
      4. Call back the `maker` so they can update their offers.
     
    **Let the devs know about any error, typo etc, by contacting 	devs@mangrove.exchange**
+
+
+  There is one Mangrove contract that manages all tradeable offerLists. This reduces deployment costs for new offerLists and lets market makers have all their provision for all offerLists in the same place.
+
+   The interaction map between the different actors is as follows:
+   <img src="./contactMap.png" width="190%"></img>
+
+   The sequence diagram of a market order is as follows:
+   <img src="./sequenceChart.png" width="190%"></img>
+
+
+   ## `tickSpacing`
+
+   The granularity of available price points in an offer list is controlled by the `tickSpacing` parameter. With a high `tickSpacing`, fewer price points are available, but the gas cost of market orders is lower since a smaller part of the tick tree has to be explored.
+
+   The available prices in an offer list are `1.0001^i` for all `MIN_TICK <= i <= MAX_TICK` such that `i % tickSpacing = 0`.
+
+   ## Tree storage
+
+   The exposed structure of offers is a list, but offers are actually stored in a tree we call a "tick tree". 
+
+   Thanks to the tree structure, offer insertion takes constant time (the height of the tree is fixed).
+
+   ### Bins
+
+   <img src="./bin.png" width="80%"></img>
+
+   Below the bottom of the tree are _bins_. A bin is a doubly-linked list of offers. All offers in a bin have the same tick. During a market order, offers in a bin are executed in order, from the first to the last. Inserted offers are always appended at the end of a bin. 
+
+   Bins are laid in sequence. If all the offers in a bin have tick `t`, all the offers in the following bin have tick `t+tickSpacing`.
+
+   Bins are identified by their index in the bin sequence, from the first (`MIN_BIN`) to the last (`MAX_BIN`). The number of available bins is larger than the number of available ticks, so some bins will never be used.
+
+   ### Tree structure
+
+   The structure of the tick tree is as follows:
+   <img src="./tick_tree_structure.png" width="190%"></img>
```

Comment:
There's a typo: "bet bin's" -> "best bin's".

Also, I think it would be good to show the offers of a bin in the overview diagram, eg something like this:

![billede](https://github.com/mangrovedao/mangrove-core/assets/3458856/6071c36e-fb70-4279-b577-25e55f083a98)